### PR TITLE
Merge Product Listings to master

### LINKED ISSUE #4597: [mobile ux] 5. Product list
## What is the problem we are solving 
Currently the list of products on the shop looks like an homage to mid 20th century brutalist soviet architecture. There is nothing about the product view that screams "BUY ME!" It's more of a "you must read and use your own imagination to fill in the gaps and work hard till you die" kinda deal.

Problems with this: 
- Photos sell products. And yet, photos hidden behind a click.
- Product details are in a modal. Modals are frowned upon on mobile. They also offer no way to allow for direct linking to a product in the shop.
- You have to type in the quantity you want, no easy, single click way to add the product to the cart.

## Success factors = expected outcome 
- [ ] Products have as much detail about them shown on the listing itself. 
- [ ] Increased cart size on mobile due to photo / product attractiveness. (can we measure this @RachL?)

## In/out of scope for phase 1
Out: individual product pages; In: maintaining current modal for individual product pages.

Out: direct linking to products; In: direct link to shop product list.

## Design Overview
**MOBILE**
Mobile: Shop - Product List

Mobile: Shop - Product Variants

**TABLET**
Tablet: Shop - Product List

**DESKTOP**
Desktop: Shop - Filters

Design details are kept in our Zeplin account: https://app.zeplin.io/project/59532764a6c78ff3c534b372/dashboard

### PR DESCRIPTION
**Pull request by mkllnk**
#### What? Why?

Closes #4597.

This is the mobile product listing feature branch. It has been reviewed in parts already and doesn't need to be reviewed in detail. One quick review is enough.

#### What should we test?

All changes have been tested already. For this PR, we should just do a sanity check:

- Go to a shopfront.
- Observe the new design. :+1: 
- Check out.

#### Release notes

Changelog Category: User facing changes

The product listing in the shopfront is has been redesigned. Images now appear on mobile. You can add items to your cart with one click. Bulk buy should be easier to understand. And the colours have been updated. :bouquet: 

:warning: **Requires a manual step before deployment:** The shopfront shows larger images now. Please read the description of https://github.com/openfoodfoundation/openfoodnetwork/issues/6236 for instructions how to resize your current images.

#### Dependencies

https://github.com/openfoodfoundation/openfoodnetwork/issues/6236 needs to be completed before we can deploy this.

#### Documentation updates

The user guide probably contains images which need updating.

### CODE REVIEW

**Comment by Matt-Yorkley** [comment]
Okay, I fixed the broken specs issue. 

I've also got a couple of small tweaks here that would be good to get some product/design feedback on (I can remove them):

#### Update placeholder image size to match new image sizes (before and after)
Screenshot from 2020-10-22 10-01-54
Screenshot from 2020-10-22 10-02-07

#### Add a slight hover effect to the images

HOVER

If an element is interactive and clicking on it will do something (as opposed to a static image), it's nice to convey that when rolling over it with the mouse. Only affects desktop view, not smaller devices.

**Comment by lin-d-hop** [comment]
Looks good @Matt-Yorkley

**Comment by Erioldoesdesign** [comment]
@Matt-Yorkley these tweaks look great to me. Especially the image size matching the new size.

Re. hover effect on images, again, makes sense and I can see the slight colour change. I have (mostly 20/20 vision). I would just flag that anyone with sight problems won't see this. But there's no essential information hidden behind the click on the image now right?

**Comment by Matt-Yorkley** [comment]
Clicking it opens a modal with more product info. In the current branch there's no hover effect at all.

**Comment by Erioldoesdesign** [comment]
re. uncontroversial totally understandable and I approve 🙌 

My assumption is that most shoppers add to cart with the information thats available via the product list and that users not satisfied are more likely to click/tap around so I'm not too worried 👍

**Review by luisramos0 — APPROVED** [review]
I went through the code changes again, it all looks good :+1:

**Comment by luisramos0** [comment]
the build doesnt look good, but maybe just a case of extreme flakyness since the select2 helper changes...

File: app/assets/stylesheets/darkswarm/_shop-inputs.scss [hunk 1]
```diff
@@ -6,93 +6,110 @@
...
+  // Override foundation button styles:
+  font-size: 1rem;
+  margin: 0;
+  padding: 0;
+  transition: none;
+
+  &:hover {
+    background-color: $orange-600;
+  }
+  &[disabled] {
```
Comment (luisramos0) [inline app/assets/stylesheets/darkswarm/_shop-inputs.scss:32]: there are some issues in code climate we could fix like here:
"Rule declaration should be followed by an empty line"

Comment (Matt-Yorkley) [reply app/assets/stylesheets/darkswarm/_shop-inputs.scss:32]: This happened with previous mobile PRs as well, `codeclimate` doesn't run on non-master branches...

**Comment by Matt-Yorkley** [comment]
Green build :heavy_check_mark:

**Comment by filipefurtad0** [comment]
Hey @Matt-Yorkley ,

After you made the resizing of the images according to #6236 and staged the PR, I checked whether:

- images were appearing correcty and with the expected size 227x227 pixels
- verified again the changes proposed in the [google-doc](https://docs.google.com/document/d/1WZZSuMCLHiZ_UY_nLJZ8MI-HGadTQmVty7zXeS7H6VE/edit#heading=h.k0m93bb41jls)
- checked the changes you introduced as well:
i) resizing of images
ii) light hover

All looked fine.

After these checks I went on to follow the testing notes,


Checked this PR on

- desktop Firefox 81 / Chrome Version 86

Browserstack:
- iPad Air 7 / Chrome

image

- iPhone 11 Pro / Safari

image

- My own desktop (Ubuntu 20.04 Firefox 81)

For reference, before (left, staging-ES) and after this PR (right, staging-AU)

image

In general, I think everything looks ok. Some observations I feel I should flag here:

- this shop looks like this right now
image

Notice:
a) the broken images - I'm not sure these were not broken before, on this server
b) the OC selector - not introduced by this PR, observable as well on staging-ES

- Error on first attempt to checkout, after this PR was deployed

The red banner appeared on check out (I don't have the pic and could not reproduce it, but it was the usual "Something went wrong with processing your order, please try again later) and triggered:
Bugsnag - https://app.bugsnag.com/yaycode/openfoodnetwork/errors/5f0e53d6186f5e0017de31b3?event_id=5f91bf2d00615385138b0000&i=sk&m=nw
BugsnagJS - https://app.bugsnag.com/yaycode/open-food-network-js-australia/errors/5f45f92b9acfde001837ae2f?event_id=5f91bf2e006160a322280000&i=sk&m=nw


In summary this looks good to go :tada: 
The only major concern would be the checkout error... Thoughts?

**Comment by Matt-Yorkley** [comment]
Thanks @filipefurtad0! :heart:

#### Elmore Organics images

None of those images appear to exist (in S3) in _any_ format, so i think it's a pre-existing issue (missing images on staging servers are pretty common). I might try staging the previous release and seeing if they reappear (just to be sure) but I think it's unrelated.

#### Order Cycle selector

I fixed it this morning, it's here: https://github.com/openfoodfoundation/openfoodnetwork/issues/6234 :wink:

#### Error on first attempt to checkout

I'll have to take a look at that, but I think the chances it's related to this PR are roughly zero :+1:
